src/components/UserSettings/UserSettings.js

```diff
@@ -1,13 +1,34 @@
-import React from "react";
+import React, { useState } from "react";
 import { connect } from 'react-redux';
 
 function UserSettings(props) {
+  const [ idleQueue, setIdleQueue] = useState(false);
+  const [ speakVoice, setSpeakVoice] = useState(true);
+  const [ focusMode, setFocusMode] = useState(false);
+
   if (!props.userSettings) {
     return <></>;
   }
 
   return (
-    <a href={props.userSettings.spotify.authorizationUrl}>Connect Your Spotify Account</a>
+    <div>
+      <a href={props.userSettings.spotify.authorizationUrl}>Connect Your Spotify Account</a>
+
+      <form>
+        <label>
+          <input type="checkbox" name="idleQueue" checked={idleQueue} onChange={() => setIdleQueue(!idleQueue)}/>
```

Comment:
My bad - I don't think I specified where this data should be stored.

Instead of this being `idleQueue` coming from `useState` it should come from the react store: `props.userSettings.idleQueue`.

Now for updating the state, it's going to be a little trickier than just calling `setIdleQueue`. You'll need to call a function, something like as follows:

```
props.dispatch({ type: "userSettings/update", payload: { idleQueue: idleQueue } })
```

Then, you'll need to add a new case to the main Redux dispatcher. You'll want to add something like this here: https://github.com/0x213F/jukebox-radio-react/blob/main/src/utils/redux.js#L200

```
case "userSettings/update":
    userSettingsUpdate(state, action.payload);
```

Then from here, you need to define the `userSettingsUpdate` function. If you see the top of the `redux.js` file you'll see functions like this one imported. You're going to need to create a new file for this function. You should add something to the top of `redux.js` like this:

```
import {
  userSettingsUpdate,
} from './reducers/userSettings';
```

Then, create a `userSettings.js` file in the reducers directory. Inside it, put:

```
export const userSettingsUpdate = function(state, payload) {
    return {
        ...state,
        userSettings: {
            ...state.userSettings,
            ...payload,
        }
    }
}
```

I'm not certain that this code will work, but _I think_ it will.

Reply:
`case "userSettings/update":
    userSettingsUpdate(state, action.payload);`

for this case should it actually be 
 `return userSettingsUpdate(state, action.payload);` 
like the others? Just wanted to make sure I had it right. I don't see why you wouldn't return that value.

Reply:
Yes - it should return! Good eye